pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -226,7 +228,7 @@ impl Drop for EphemeralFile {
     fn drop(&mut self) {
         // drop all pages from page cache
         let cache = page_cache::get();
-        cache.drop_buffers_for_immutable(self.page_cache_file_id);
+        Handle::current().block_on(cache.drop_buffers_for_immutable(self.page_cache_file_id));
```

Comment:
Similar drop problem here.

I wonder if this could be made ownership based.

Reply:
discussed down below.